Ghosting boundaries (#27810)
idaholab/moose: framework/doc/content/syntax/Mesh/splitting.md

will be written to a binary file under the generated directory that can be loaded when using
split meshes.

!alert note

> The mesh splitter commands do not work with DistributedMesh. You must only split with a ReplicatedMesh.

Reviewer: ```suggestion
The mesh splitter commands do not work with a [distributed mesh](syntax/Mesh/index.md#replicated-and-distributed-mesh). You can only split starting from a replicated mesh.
```

## Using Split Meshes

> There are two ways of using referring to meshes in parallel in MOOSE, as split or as distributed. While there is no conceptual difference between the two there is a difference in implementation and features.

Reviewer: a split mesh is a mesh read from a split file so far. that's the conceptual difference

Author: Not sure it's entirely so given they both handle distributed meshes and there is an overlap, but if you'd like to settle on this difference for MOOSE I am happy with it.

Reviewer: it is the difference. If you dont read it from a pre-split file it's not a split mesh

> There are two ways of using referring to meshes in parallel in MOOSE, as split or as distributed. While there is no conceptual difference between the two there is a difference in implementation and features.

Reviewer: ```suggestion
There are two ways of referring to distributed meshes in MOOSE, as split or as distributed. A split mesh is a distributed mesh. A split mesh is read from a pre-split mesh file (from a [Checkpoint.md] or using the instructions above). A distributed mesh can be generated in parallel or loaded entirely replicated then distributed.
```

### Loading a split mesh using [FileMeshGenerator.md]

> A split mesh, i.e. a `.cpr` file, may be loaded using a [FileMeshGenerator.md] if access to the mesh is needed before the full simulation case is set-up. In fact this approach is recommended; it leads to correct element ghosting whereas the approach we introduce in the following section may not.

Reviewer: ```suggestion
A split mesh, i.e. a `.cpr` file, may be loaded using a [FileMeshGenerator.md] if access to the mesh is needed before the full simulation case is set-up, notably for additional mesh generation with [mesh generators](syntax/Mesh/index.md#mesh-generators). In fact this approach is recommended; it leads to correct element ghosting whereas the approach we introduce in the following section may not.
```

Reviewer: you ll have to add a `id=mesh-generators` to syntax/Mesh/index.md right near the beginning to make that work

$ mpiexec -n 42 moose-app-opt -i your_input.i --use-split --split-file foo.cpr
```

!alert warning

> The `--use-split` option skips entirely the `[Mesh]` block of the input file, and as such some data structure initializations required if further mesh operations are necessary at the simulation start-up stage. 

Author: I actually like the alert style reminder, would you be ok to keep it even if it seems repetitive?

> The `--use-split` option skips entirely the `[Mesh]` block of the input file, and as such some data structure initializations required if further mesh operations are necessary at the simulation start-up stage. 

Reviewer 2: yea that's fine
